lib/Onyx.js

```diff
@@ -380,6 +381,17 @@ function keyChanged(key, data) {
 
         // Subscriber is a regular call to connect() and provided a callback
         if (_.isFunction(subscriber.callback)) {
+            // The subscriber here is only subscribing to a particular piece of the data, so we must only trigger
+            // the callback if that particular piece of data has changed
+            if (subscriber.selector) {
+                const previousValue = lodashGet(previousData, subscriber.selector);
+                const newValue = lodashGet(data, subscriber.selector);
```

Comment:
I think those are good concerns. Let's go with the simple option of using strings for now, we can add functions later if we think they are useful.

Reply:
Hmm but why choose simple over useful? I say go with the thing that makes sense for how we will use it. We can pick 5 places where this might be useful and see if it solves the problem we think exists...

To me, a problem could also be:

1. We are subscribing to a big object of data
2. Something small changes about the data
3. We care about everything, but the thing that changed
4. Whole thing updates anyway

With a selector you don't have this problem since you can omit what you don't want and return that back to the component. With something like this path selector idea you will have to add n subscribers for every field that you want.

Reply:
Oh not saying we should go with simple over useful. I want useful AND simple. So let's see what's the best way to get both.
Also, maybe let's keep this discussion in [this other thread](https://github.com/Expensify/react-native-onyx/pull/192#discussion_r990069098) so we are not repeating things